dubbo-config/dubbo-config-api/src/main/java/com/alibaba/dubbo/config/AbstractConfig.java

```diff
@@ -71,6 +71,8 @@
         legacyProperties.put("dubbo.consumer.retries", "dubbo.service.max.retry.providers");
         legacyProperties.put("dubbo.consumer.check", "dubbo.service.allow.no.provider");
         legacyProperties.put("dubbo.service.url", "dubbo.service.address");
+
+        Runtime.getRuntime().addShutdownHook(DubboShutdownHook.getDubboShutdownHook());

```

Comment:
This will lead to the coexistence of spring hooks，If the spring hook destroys the context before dubbo, the user code uses the aop feature, or calls the getBean method, dubbo gracefully shutdown and reports that the bean was not found.
Only dubbo hooks are registered in non-spring containers.